Renamed the audit_log_api source to logging_log_entry and implement the Retry mechanism

ParthaI opened the pull request:
# Example query results
<details>
  <summary>Results</summary>

```
Add example SQL query results here (please include the input queries as well)
```
</details>

ParthaI commented:
## Summary of Changes

This PR migrates the GCP logging log entry source from the legacy `cloud.google.com/go/logging/logadmin` client to the modern `cloud.google.com/go/logging/apiv2` (GAPIC) client and implements a configurable retry mechanism with exponential backoff.

### Key Changes:

**1. Client Migration**
- Replaced `cloud.google.com/go/logging/logadmin` with `cloud.google.com/go/logging/apiv2` (GAPIC client)
- Updated API calls from `client.Entries()` to `client.ListLogEntries()`
- Modified data mapping to handle protobuf structures (`loggingpb.LogEntry`, `audit.AuditLog`)
- Updated field access patterns to use protobuf getter methods

**2. Retry Mechanism Implementation**
- Added configurable retry parameters at the connection level (`config/gcp_connection.go`):
  - `min_retry_delay` (Initial delay, default: 500ms)
  - `max_retry_delay` (Maximum delay, default: 60s) 
  - `backoff_multiplier` (Exponential growth factor, default: 1.30)
- Implemented exponential backoff with jitter using `gax.WithRetry` and `gax.Backoff`
- Added retry for specific error codes: `Unavailable`, `DeadlineExceeded`, `ResourceExhausted`, `Aborted`

**3. Data Mapping Updates**
- Updated `tables/audit_log/audit_log_mapper.go` to handle both GAPIC and logadmin client data formats
- Fixed protobuf payload extraction using `proto.Unmarshal` for `anypb.Any` fields
- Corrected HTTP request field mapping for protobuf structures
- Maintained backward compatibility for existing log data

**4. Configuration Improvements**
- Aligned parameter naming with GCP SDK defaults
- Added validation for retry parameters

**5. Documentation Updates**
- Updated source documentation to reflect new retry configuration
- Updated example configurations to show connection-level retry settings

### Benefits:
- **Improved Reliability**: Robust retry mechanism handles transient network and service errors
- **Better Performance**: GAPIC client provides improved performance and features
- **Enhanced Maintainability**: Modern client with better long-term support
- **Configurable Resilience**: Users can customize retry behavior based on their needs

### Breaking Changes:
- The source name is only the breaking changes.

ParthaI commented on sources/logging_log_entry/logging_log_entry_source.go:
```diff
@@ -0,0 +1,232 @@
+		filters := s.logTypeFilter.GetLogFiltersForTable(tableName, projectId)
+		allFilters = append(allFilters, filters...)
+	}
+	return fmt.Sprintf("logName=(%s) %s", strings.Join(allFilters, " OR "), timePart)
```
Not required

ParthaI commented on sources/logging_log_entry/logging_log_entry_source.go:
```diff
@@ -0,0 +1,232 @@
+		}
+		if err != nil {
+			// If we get here, retries were already applied & exhausted.
+			return fmt.Errorf("ListLogEntries failed after retries: %w", err)
```
The retry mechanism has been handle as per the instructions provided in this doc: https://cloud.google.com/storage/docs/retry-strategy#customize-retries

recharge-jmog commented:
@ParthaI I rebased my local PR branch with this PR and did a little testing.  Here's what I've seen:  

- I was able to pull over 2M logs from the Cloud Logging API with no rate errors, so that seems to be working well
- I received this error on 2 rows:  2 rows with error: error decoding service data: unsupported type: type.googleapis.com/google.cloud.bigquery.logging.v1.AuditData indicating there may be some addition needed to get Big Query audit logs supported


So overall, looking pretty good.  I presume the use pattern will be just setting up different partitions for the other log types (requests, etc) and collecting them separately?

ParthaI commented:
Hello @recharge-jmog,

Thank you for testing the changes and sharing your feedback — it's much appreciated!

* I’ve addressed the `error decoding service data` issue, so this error should no longer occur going forward.

> *"So overall, looking pretty good. I presume the use pattern will be just setting up different partitions for the other log types (requests, etc) and collecting them separately?"*

Yes, exactly. Each log type (e.g., request logs, etc.) will be handled via its own **table and partition**. In the original PR, I had reverted the changes related to the `request_logs` table, but you can refer to [this commit](https://github.com/turbot/tailpipe-plugin-gcp/pull/93/commits/05f1c2fbc709d790340f8e4a89ded4d75f8e18cf) as an example of how it was structured earlier.

Thanks again!

ParthaI commented:
Yes, correct @recharge-jmog!
